### PR TITLE
 Add scenario id (EXPOSUREAPP-11771)

### PR DESCRIPTION
**Pull request by chiljamgossow**
Provide admission scenario id for `DccWalletCalculation`

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/DccWalletInfoCalculationManager.kt [hunk 1]
```diff
@@ -34,32 +38,28 @@ class DccWalletInfoCalculationManager @Inject constructor(
                 person.dccWalletInfo == null ||
                 person.dccWalletInfo.validUntilInstant.isBefore(now)
             ) {
-                updateWalletInfoForPerson(person)
+                updateWalletInfoForPerson(person, admissionScenarioId)
             }
         }
+        cclSettings.setAdmissionScenarioId(admissionScenarioId)
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/DccWalletInfoCalculationManager.kt:44]: That should not happen here anymore

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/DccWalletInfoCalculationManager.kt [hunk 2]
```diff
@@ -19,12 +19,16 @@ class DccWalletInfoCalculationManager @Inject constructor(
     private val dccWalletInfoRepository: DccWalletInfoRepository,
     private val calculation: DccWalletInfoCalculation,
     private val timeStamper: TimeStamper,
+    private val cclSettings: CCLSettings,
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/DccWalletInfoCalculationManager.kt:22]: There is no need to inject it here

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/DccWalletInfoCalculationManagerTest.kt [hunk 3]
```diff
@@ -37,6 +38,7 @@ class DccWalletInfoCalculationManagerTest : BaseTest() {
     @MockK lateinit var certificatePersonIdentifier2: CertificatePersonIdentifier
     @MockK lateinit var dccWalletInfo1: DccWalletInfo
     @MockK lateinit var dccWalletInfo2: DccWalletInfo
+    @MockK lateinit var cclSettings: CCLSettings
```
Comment (mtwalli) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/DccWalletInfoCalculationManagerTest.kt:41]: should be removed from the class and therefore the test

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/DccWalletInfoCalculationManager.kt [hunk 4]
```diff
@@ -34,32 +36,26 @@ class DccWalletInfoCalculationManager @Inject constructor(
...
         }
+        Result.Success
     } catch (e: Exception) {
         Timber.e(e, "Failed to run calculation.")
-    }
-
-    /**
-     * Trigger [DccWalletInfo] calculation for specific person
-     */
-    suspend fun triggerCalculationForPerson(personIdentifier: CertificatePersonIdentifier) {
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/DccWalletInfoCalculationManager.kt:56]: Why was this removed? Also why was the option to re-calculate the wallet for a single person in the Test Menu removed?

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/DccWalletInfoCalculationManager.kt:56]: Test menu for this option was removed too

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/update/DccWalletInfoUpdateTrigger.kt [hunk 5]
```diff
@@ -9,35 +11,51 @@ import timber.log.Timber
...
             )
         )
     }
 
+    private suspend fun getAdmissionScenarioId(): String {
+        val disabled = runCatching {
+            appConfigProvider.getAppConfig().admissionScenariosDisabled
+        }.onFailure {
+            Timber.d(it, "getAppConfig().admissionScenariosDisabled failed")
+        }.getOrElse { true }
```
[the remark this answers is not in the document]
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/update/DccWalletInfoUpdateTrigger.kt:54]: this is `admissionScenariosDisabled` was there a change in the specs afaik it was` _DISABLED` ,but you are right we should be consistent

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/DccWalletInfoCalculationManager.kt (hunk 4 above)
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/DccWalletInfoCalculationManager.kt:56]: okay.

**Review by LukasLechnerDev — DISMISSED** [review]
LGTM

**Review by LukasLechnerDev — APPROVED** [review]

**Review by mtwalli — APPROVED** [review]
